contribute/pull-request.md

```diff
@@ -0,0 +1,99 @@
+
+If the pull request fixes a bug:
+
+- Pull request description must include `Closes #<issue>` or `Fixes #<issue>`.
+- Pull request should include tests that replicate the fixed bug, to avoid regressions.
+
+### Frontend-specific guidelines
+
+- Pull request doesn't increase the Angular code base.
+- Pull request doesn't contain uses of `any` or `{}` without comments describing why.
+- Pull request doesn't contain large React components that could easily be split into several smaller components.
+- Pull request doesn't contain backend calls directly from components—use actions and Redux instead.
+- Pull request uses [Emotion](https://github.com/grafana/grafana/blob/master/style_guides/styling.md) for styling.
+
+#### Redux
+
+- Pull request doesn't contain code that mutates state in reducers or thunks.
+- Pull request uses the `actionCreatorFactory` and `reducerFactory` helpers instead of traditional switch statement reducers in Redux. Refer to [Redux framework](https://github.com/grafana/grafana/tree/master/style_guides/redux.md) for more details.
+- Pull request uses `reducerTester` to test reducers. Refer to [Redux framework](https://github.com/grafana/grafana/tree/master/style_guides/redux.md) for more details.
+- Pull request doesn't contain code that accesses the reducers state slice directly. Instead, the code should use state selectors to access state.
+
+## Code review
+
+Once you've created a pull request, the next step is to have someone review your change. A review is a learning opportunity for both the reviewer and the author of the pull request.
+
+If you think a specific person needs to review your pull request, you can tag them in the description or in a comment. Tag a user by typing the `@` symbol followed by their GitHub username.
+
+We recommend that you read [How to do a code review](https://google.github.io/eng-practices/review/reviewer/) to learn more about some best practices for code reviews.
+
+## Style guide
+
+A well-written pull request minimizes the time to get your change accepted. This style guide describes how to write good commit messages and descriptions for your pull requests.
+
+### Commit message format
+
+Grafana uses the guidelines for commit messages outlined in [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/), with the following additions:
+
+- Subject line must begin with the _area_ of the commit.
+- A footer in the form of an optional [keyword and issue reference](https://help.github.com/en/articles/closing-issues-using-keywords).
+
+#### Area
+
+Should be one of the following:
+
+- **Build**: Changes to the build system, or external dependencies.
+- **Chore**: Changes that don't affect functionality.
+- **Dashboard**: Changes to the Dashboard feature.
+- **Docs**: Changes to documentation.
+- **Explore**: Changes to the Explore feature.
+- **Plugins**: Changes to any of the plugins.
+
+Changes to data sources:
+
+- AzureMonitor
```

Comment:
I know these are not the only three data sources, so are they examples? Unclear.